nbbuild/jenkins/github/Jenkinsfile.groovy

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+ 
+pipeline {
+    agent { 
+        node { 
+            label 'ubuntu' 
+        }
+    }
+    
+    tools {
+        jdk 'JDK 1.8 (latest)'
+        ant 'Ant (latest)'        
+    }
+
+    triggers {
+        issueCommentTrigger('.*test this please.*')
+    }
+ 
+    environment { 
+        ANT_OPTS = '-Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json'
+    }
+
+    stages {
```

Comment:
Can't these run in parallel?

Reply:
To be useful, that would require separate checkouts, and separate agents.

Since I presume `commit-validation` is by far the longest part, it would not buy you much to parallelize.

Reply:
Oddly, seems that running sigtests takes _longer_ than commit validation. Seems like something is wrong there.

Reply:
Remember that if you break up targets into `stage`s, you are potentially rerunning the same dependent targets and preventing Ant from using its dependency graph effectively. Generally recommend minimizing the total number of `sh` steps if possible: ideal to do the whole build in one line. At least compare times to see if that is contributing overhead.

Reply:
I prefer to run ant tasks in stages for a better debug when it fails but I change the order to prevent run build task twice.
The other tasks doesn't depend to buid task.

Reply:
Build task is required but is only executed in his step, the other tasks doesn't rerun any task except bootstrap or init task.